source/common/filter/config_discovery_impl.h

```diff
@@ -29,6 +29,13 @@ class FilterConfigSubscription;
 
 using FilterConfigSubscriptionSharedPtr = std::shared_ptr<FilterConfigSubscription>;
 
+// These helper functions throw exceptions. Thus can not be defined in .h files.
+void validateProtoConfigDefaultFactoryHelper(const bool null_default_factory,
```

Comment:
Try having functions which are scoped inside a class or are in an empty namespace.
The helper functions should probably be either private or protected functions in one of the classes below.

Reply:
ACK